test/integration/users.test.ts

```diff
@@ -0,0 +1,39 @@
+import chai from 'chai';
+
+import app from '../../server';
+import { generateAuthToken } from '../../services/authService';
+const { expect } = chai;
+
+describe('Integration | Route | /users/self', function () {
+  it('Get | /user/self | Error', function (done) {
+    chai
+      .request(app)
+      .get('/api/v1/users/self')
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+        expect(res).to.have.status(401);
+        expect(res.body.error.message).equal('Unauthenticated User');
+        expect(res.body.error.statusCode).equal(401);
+        expect(res.body.error.error).equal('Unauthorized');
+        return done();
+      });
+  });
+
+  it('GET | /user/self | Success', function (done) {
+    const jwt = generateAuthToken({ userId: 12524395 });
```

Comment:
can we mock this?

Reply:
generate auth token ins an internal function why do we need to mock this